app/inpututils.cpp

```diff
@@ -111,6 +112,70 @@ QString InputUtils::formatNumber( const double number, int precision )
   return QString::number( number, 'f', precision );
 }
 
+QString InputUtils::formatDuration( int lengthOfCycle, const QString &cycleName )
+{
+  if ( lengthOfCycle == 0 || lengthOfCycle == 1 )
+  {
+    return QStringLiteral( "%1 %2 ago" ).arg( lengthOfCycle ).arg( cycleName );
+  }
+  else
+  {
+    return QStringLiteral( "%1 %2s ago" ).arg( lengthOfCycle ).arg( cycleName );
```

Comment:
This is not very translation friendly. Also, I think the correct english version for zero is "0 hours ago", not "0 hour ago"

Qt has support for plural forms in tr() - https://doc.qt.io/qt-5/i18n-source-translation.html#handling-plurals - some languages have multiple plural forms - but not sure if we support it with transifex + we may need to add english translation (to convert generic  "%1 week(s) ago" to "%1 week ago" / "%1 weeks ago" depending on the number. So maybe as a compromise let's stick to `if (length > 1) tr( "%1 weeks ago") else tr("%1 week ago")`. Let's have multiple texts ("%1 day ago", "%1 days ago", "%1 week ago", ...) - it will give better context to translators.